packages/wes_api/client/api_workflow_execution_service_log_extension.go

```diff
@@ -0,0 +1,146 @@
+ */
+
+package wes_client
+
+import (
+	_context "context"
+	_ioutil "io/ioutil"
+	_nethttp "net/http"
+	_neturl "net/url"
+	"fmt"
+	"strings"
+)
+
+// Linger please
+var (
+	_ _context.Context
+)
+
+/*
+GetRunLogData Get data linked to by GetRunLog.
+Returns the content of a URL reverenced in a GetRunLog response, if that URL is also on the WES server.
+ * @param ctx _context.Context - for authentication, logging, cancellation, deadlines, tracing, etc. Passed from http.Request or context.Background().
+ * @param runId The run ID used for the GetRunLog call
+ * @param dataUrl The URL string from the GetRunLog call
+@return string
+*/
+func (a *WorkflowExecutionServiceApiService) GetRunLogData(ctx _context.Context, runId string, dataUrl string) (string, *_nethttp.Response, error) {
+	var (
+		localVarHTTPMethod   = _nethttp.MethodGet
+		localVarPostBody     interface{}
+		localVarFormFileName string
+		localVarReturnValue  string
+	)
+
+	// create path and map variables
+	localVarPath := a.client.cfg.BasePath + "/runs/{run_id}"
+	localVarPath = strings.Replace(localVarPath, "{"+"run_id"+"}", _neturl.PathEscape(parameterToString(runId, "")), -1)
+
+	// Evaluate dataUrl relative to localVarPath and replace localVarPath
+	base, err := _neturl.Parse(localVarPath)
+    if err != nil {
+        return localVarReturnValue, nil, err
+    }
+	evaluated, err := base.Parse(dataUrl)
+	if err != nil {
+        return localVarReturnValue, nil, err
+    }
+	if (evaluated.Scheme != base.Scheme && !strings.HasPrefix(evaluated.Scheme, "http")) {
+		// This doesn't look like something we can fetch
+		return localVarReturnValue, nil, fmt.Errorf("WES cannot be used to retrieve %s", dataUrl)
+	}
+	localVarPath = evaluated.String()
+
+	localVarHeaderParams := make(map[string]string)
+	files := make(map[string][]byte)
+	localVarQueryParams := _neturl.Values{}
+	localVarFormParams := _neturl.Values{}
+
+	// to determine the Content-Type header
+	localVarHTTPContentTypes := []string{}
+
+	// set Content-Type header
+	localVarHTTPContentType := selectHeaderContentType(localVarHTTPContentTypes)
+	if localVarHTTPContentType != "" {
+		localVarHeaderParams["Content-Type"] = localVarHTTPContentType
+	}
+
+	// to determine the Accept header
+	localVarHTTPHeaderAccepts := []string{"text/plain"}
+
+	// set Accept header
+	localVarHTTPHeaderAccept := selectHeaderAccept(localVarHTTPHeaderAccepts)
+	if localVarHTTPHeaderAccept != "" {
+		localVarHeaderParams["Accept"] = localVarHTTPHeaderAccept
+	}
+	r, err := a.client.prepareRequest(ctx, localVarPath, localVarHTTPMethod, localVarPostBody, localVarHeaderParams, localVarQueryParams, localVarFormParams, localVarFormFileName, files)
+	if err != nil {
+		return localVarReturnValue, nil, err
+	}
+
+	localVarHTTPResponse, err := a.client.callAPI(r)
+	if err != nil || localVarHTTPResponse == nil {
+		return localVarReturnValue, localVarHTTPResponse, err
+	}
+
+	localVarBody, err := _ioutil.ReadAll(localVarHTTPResponse.Body)
```

Comment:
Are we concerned about using `ReadAll` to load a potentially-large log file?

Reply:
I'm not overly concerned about file size here; even a very large workflow log is probably under a gigabyte.

Streaming would of course be better, but I didn't want to invest the time to learn, design and implement a good way to stream it here. It's going to the console, so when it starts to get too large for memory it starts to get too large to really be useful. But it's also not a good idea to let a very large file crash the client.

(*Really* we would want to be able to make HEAD and byte range requests or something, and be able to get just a manageable piece from the *end* of the log, and to let the user then ask for other parts of the log somehow, but I also didn't want to build out that whole system right now.)

Maybe implementing streaming wouldn't actually be more work than it's worth. I could revise the download code to be able to handle errors internally while streaming successful responses, and change the return type to whatever type `Body` is, and then where it's actually used figure out what to do with the stream (it looks like `io.Copy` can copy streams to streams?).

Reply:
Adjusting this to return a stream, and passing the stream up to the CLI command implementation in `logs_workflow.go`, and using `io.Copy` there, breaks the tests, because the data is copied to the standard output of the test process, instead of wherever the output capturing machinery that makes the `expectedOutput` values in `logs_workflow_test.go` expects the output to be going. It looks like the test code allows only `printLn` to be used for output.